### PR TITLE
[Enhencement](tvf) select tvf supports using resource

### PR DESCRIPTION
**Pull request by BePPPower**
## Proposed changes

Issue Number: close #xxx

Create an S3/HDFS resource that TVF can use it directly to access the data source.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run p0

**Review by morningman — COMMENTED** [review]
We missed the authorization issue

File: fe/fe-core/src/main/java/org/apache/doris/tablefunction/ExternalFileTableValuedFunction.java [hunk 1]
```diff
@@ -174,9 +175,18 @@ protected void parseFile() throws AnalysisException {
 
     //The keys in properties map need to be lowercase.
     protected Map<String, String> parseCommonProperties(Map<String, String> properties) throws AnalysisException {
+        Map<String, String> mergedProperties = Maps.newHashMap();
+        if (properties.containsKey("resource")) {
+            Resource resource = Env.getCurrentEnv().getResourceMgr().getResource(properties.get("resource"));
+            if (resource == null) {
+                throw new AnalysisException("Can not find resource: " + properties.containsKey("resource"));
```
[the remark this answers is not in the document]
Comment (BePPPower) [reply fe/fe-core/src/main/java/org/apache/doris/tablefunction/ExternalFileTableValuedFunction.java]: done

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run external performance

**Comment by BePPPower** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM
But miss authorization check, will be fixed in next PR

**Review by zddr — APPROVED** [review]

**Pull request merged**